api/lcp/server.py

```diff
@@ -0,0 +1,351 @@
+        """Returns a Hasher instance
+
+        :param configuration: Configuration object
+        :type configuration: LCPServerConfiguration
+
+        :return: Hasher instance
+        :rtype: hash.Hasher
+        """
+        if self._hasher_instance is None:
+            self._hasher_instance = self._hasher_factory.create(
+                configuration.encryption_algorithm)
+
+        return self._hasher_instance
+
+    def _create_partial_license(self, configuration, patron_id=None, license_start=None, license_end=None):
+        """Creates a partial LCP license used an input by the LCP License Server for generation of LCP licenses
+
+        :param configuration: Configuration object
+        :type configuration: LCPServerConfiguration
+
+        :param patron_id: Patron's unique ID
+        :type patron_id: Optional[string]
+
+        :param license_start: Date and time when the license begins
+        :type license_start: Optional[datetime.datetime]
+
+        :param license_end: Date and time when the license ends
+        :type license_end: Optional[datetime.datetime]
+
+        :return: Partial LCP license
+        :rtype: Dict
+        """
+        hasher = self._get_hasher(configuration)
+        encrypted_password = hasher.hash(self.DEFAULT_PATRON_PASSPHRASE)
+        partial_license = {
+            'provider': configuration.provider_name,
+            'encryption': {
+                'user_key': {
+                    'text_hint': configuration.passphrase_hint,
+                    'hex_value': encrypted_password,
+                }
+            },
+            'rights': {
+                'print': int(configuration.max_printable_pages),
+                'copy': int(configuration.max_copiable_pages)
+            }
+        }
+
+        if patron_id:
+            partial_license['user'] = {
+                'id': str(patron_id)
+            }
+        if license_start:
+            partial_license['rights']['start'] = utils.format_datetime(license_start)
+        if license_end:
+            partial_license['rights']['end'] = utils.format_datetime(license_end)
+
+        return partial_license
+
+    @staticmethod
+    def _send_request(configuration, method, path, payload, json_encoder=None):
+        """Sends a request to the LCP License Server
+
+        :param configuration: Configuration object
+        :type configuration: LCPServerConfiguration
+
+        :param path: URL path part
+        :type path: string
+
+        :param payload: Dictionary containing request's payload (should be JSON compatible)
+        :type payload: Union[Dict, object]
+
+        :param json_encoder: JSON encoder
+        :type json_encoder: JSONEncoder
+
+        :return: Dictionary containing LCP License Server's response
+        :rtype: Dict
+        """
+        json_payload = json.dumps(payload, cls=json_encoder)
+        url = urlparse.urljoin(configuration.lcpserver_url, path)
+        response = requests.request(
```

Comment:
Usually we make the HTTP request in a method that does nothing else, so we can mock it during tests. But based on your changes to requirements.txt I'm guessing you mock the requests library itself.

Reply:
Yes, it's much easier to mock using `requests-mock` library. For example:
```python
            with requests_mock.Mocker() as request_mock:
                url = urlparse.urljoin(fixtures.LCPSERVER_URL, '/contents/{0}/license'.format(fixtures.CONTENT_ID))
                request_mock.post(url, json=fixtures.LCPSERVER_LICENSE)
```